src/dbnode/storage/namespace_readers.go

```diff
@@ -199,6 +205,22 @@ func (m *namespaceReaderManager) filesetExistsAt(
 		m.namespace.ID(), shard, blockStart, latestVolume)
 }
 
+func (m *namespaceReaderManager) assignShardSet(shardSet sharding.ShardSet) {
+	m.Lock()
+	defer m.Unlock()
+	m.shardSet = shardSet
+}
+
+func (m *namespaceReaderManager) isShardAvailableWithLock(shard uint32) bool {
+	shardState, err := m.shardSet.LookupStateByID(shard)
+	// NB(bodu): LookupStateByID returns ErrInvalidShardID when shard
+	// does not exist in the shard map which means the shard is not available.
+	if err != nil {
+		return false
+	}
+	return shardState == shardTypes.Available
```

Comment:
As with the seeker manager I believe all we need to do is check that the shard exists in the shard set or not, and if it does not then close it.